c/certificateService.c

```diff
@@ -109,110 +91,86 @@ static void respondWithInvalidMethod(HttpResponse *response) {
     finishResponse(response);
 }
 
-static void respondWithBadRequest(HttpResponse *response) {
-    jsonPrinter *p = respondWithJsonPrinter(response);
-
-    setResponseStatus(response, 400, "Bad Request");
-    setDefaultJSONRESTHeaders(response);
-    writeHeader(response);
-
-    jsonStart(p);
-    {
-      jsonAddString(p, "error", "The length of the certificate is longer than 4096 bytes");
-    }
-    jsonEnd(p);
-
-    finishResponse(response);
-}
-
-static int serveMappingService(HttpService *service, HttpResponse *response)
-{
+static int serveMappingService(HttpService *service, HttpResponse *response) {
   HttpRequest *request = response->request;
 
   if (!strcmp(request->method, methodPOST))
   {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
 
-    if(request->contentLength > sizeof(userMapCertificateStructure->certificate) || request->contentLength < 0) {
-      respondWithBadRequest(response);
-      return 0;
+    int urlLength = strlen(request->uri);
+    if(urlLength < 0 || urlLength > 17) {
+        respondWithJsonError(response, "URI is longer than maximum length of 17 characters.", 400, "Bad Request");
+        return 0;
     }
 
-    userMapCertificateStructure->certificateLength = request->contentLength;
-    memset(userMapCertificateStructure->certificate, 0, request->contentLength);
-    memcpy(userMapCertificateStructure->certificate, request->contentBody, request->contentLength);
+    char translatedURL[urlLength + 1];
+    strcpy(translatedURL, request->uri);
+    a2e(translatedURL, sizeof(translatedURL));
+    char *x509URI = strstr(translatedURL, "x509");
+    char *dnURI = strstr(translatedURL, "dn");
```

Comment:
What does "dn" stand for? Distinguished name?

Reply:
yes, dn stands for distinguished name, there is full name now